packages/assets-controllers/src/TokensController.ts

```diff
@@ -767,7 +767,7 @@ export class TokensController extends BaseController<
     }
 
     let newAllDetectedTokens = allDetectedTokens;
-    if (newDetectedTokens?.length) {
+    if (newDetectedTokens !== undefined) {
```

Comment:
could you explain the reason why this solves the issue?

Reply:
When we import all the tokens from detectedTokens using addTokens, we need to clear the `allDetectedTokens[chainId][selectedAddress]` as well since we are moving the data to either tokens[] or ignoredTokens[], and it should no longer be available as detectedToken. When we skip for  `newDetectedTokens?.length <=0`, this clearing is not happening, and the data in `allDetectedTokens[chainId][selectedAddress]` remains, which is fetched during network and preference state change. So we need to update `allDetectedTokens[chainId][selectedAddress]` to an empty list to avoid this reloading.

Reply:
Sorry if I'm being dense but I'm still confused why isn't it clearing when `newDetectedTokens` has a length property but is clearing when it is simply defined? 

Also: should we also be making this change [here](https://github.com/MetaMask/controllers/blob/25b122b8326aaac0b53dcf30a7daa2276b92a7a5/packages/assets-controllers/src/TokensController.ts#L744) and [here](https://github.com/MetaMask/controllers/blob/25b122b8326aaac0b53dcf30a7daa2276b92a7a5/packages/assets-controllers/src/TokensController.ts#L757) as well?

Reply:
When we are checking for `newDetectedTokens.length` empty list`[]` was getting skipped as well, removing that ensured the empty list is also getting assigned to `allDetectedTokens[chainId][selectedAddress]`. 

I think we can update the condition checking for allTokens and allIgnoredTokens updates as well.